### PR TITLE
FOGL-7457: Fixed configuration issues

### PR DESCRIPTION
**Pull request by mshariq-nerd**
Fixed Issues

1. [FOGL-7457](https://scaledb.atlassian.net/browse/FOGL-7457)
2. [FOGL-7458](https://scaledb.atlassian.net/browse/FOGL-7458)
3. [FOGL-7477](https://scaledb.atlassian.net/browse/FOGL-7477)
4. [FOGL-7470](https://scaledb.atlassian.net/browse/FOGL-7470)
5. JSON format issue 
6. validity expression checks

### CODE REVIEW

File: src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.html [hunk 1]
```diff
@@ -53,9 +55,11 @@
                 <div class="column">
                   <div class="field">
                     <div class="control">
-                      <input class="input" id="name" type="text" placeholder="name" title="No double quotes!" autocomplete="off" formControlName="name" required
-                        (input)="validateServiceName($event)">
-                      <small *ngIf="!isValidName" class="help is-danger level-left">Application name is required</small>
+                      <input class="input" id="name" type="text" placeholder="name" title="No double quotes!"
```
Comment (praveen-garg) [inline src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.html:58]: why the required is removed?

File: src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.html [hunk 2]
```diff
@@ -129,9 +138,11 @@
             <div class="field-body">
               <div class="field">
                 <div class="control">
-                  <input class="input" id="name" type="text" placeholder="name" title="No double quotes!" autocomplete="off" formControlName="name" required
-                    (input)="validateServiceName($event)">
-                  <small *ngIf="!isValidName" class="help is-danger level-left">Application name is required</small>
+                  <input class="input" id="name" type="text" placeholder="name" title="No double quotes!"
```
Comment (praveen-garg) [inline src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.html:141]: required removed?

File: src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.ts [hunk 3]
```diff
@@ -21,10 +22,9 @@ export class AddFilterWizardComponent implements OnInit {
   public plugins = [];
   public categories = [];
   public configurationData;
-  public isValidPlugin = true;
-  public isSinglePlugin = true;
-  public isValidName = true;
-  public payload: any;
+  //public isValidPlugin = true;
+  //public isSinglePlugin = true;
+  // public isValidName = true;
```
Comment (praveen-garg) [inline src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.ts:27]: remove?

File: src/app/components/core/north/add-task-wizard/add-task-wizard.component.html [hunk 4]
```diff
@@ -119,18 +120,17 @@
                   <div class="columns is-mobile">
                     <div class="column">
                       <input class="input" type='number' appNumberOnly min="0" max="365" placeholder="days"
-                        formControlName="repeatDays" (change)="validateRepeatDays($event)">
-                      <span *ngIf="!isValidDay || (repeatDays.errors?.min || repeatDays.errors?.max)"
```
Comment (praveen-garg) [inline src/app/components/core/north/add-task-wizard/add-task-wizard.component.html:123]: logic changed?

File: src/app/components/core/north/add-task-wizard/add-task-wizard.component.html [hunk 5]
```diff
@@ -119,18 +120,17 @@
...
                         <sup>*</sup>Repeat days must be in range 0 - 365.
                       </span>
                     </div>
                     <div class="column">
                       <input class="input" type='text' placeholder="HH:mm:ss" formControlName="repeatTime"
-                        mask='00:00:00' (input)="setRepeatIntervalValue($event)" [pattern]='regExp'
-                        (change)="validateRepeatTime($event)">
+                        mask='00:00:00' (input)="setRepeatIntervalValue($event)" [pattern]='regExp'>
 
-                      <span *ngIf="!isValidTime || repeatTime.errors?.pattern" class="help is-danger">
```
Comment (praveen-garg) [inline src/app/components/core/north/add-task-wizard/add-task-wizard.component.html:133]: logic changed?

File: src/app/directives/custom-validator.ts [hunk 6]
```diff
@@ -17,4 +17,9 @@ export class CustomValidator {
       }
     });
   }
+
+  public static pluginsCountValidator(control: AbstractControl): { [s: string]: boolean } {
+    const value = control.value;
+    return value && value.length > 1 ? { multiplePlugins: true } : null;
```
Comment (praveen-garg) [inline src/app/directives/custom-validator.ts:23]: did we test when no plugin installed? for filter, rule/delivery and south?

File: src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.html (hunk 1 above)
Comment (mshariq-nerd) [reply src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.html:58]: The required is in the .ts file when we are creating form group so removed from here.

File: src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.html (hunk 2 above)
Comment (mshariq-nerd) [reply src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.html:141]: The required is in the ts file when we are creating form group so removed from here.

**Review by monika-Dianomic — DISMISSED** [review]

**Review by praveen-garg — DISMISSED** [review]

File: src/app/components/core/north/add-task-wizard/add-task-wizard.component.ts [hunk 7]
```diff
@@ -324,7 +269,17 @@ export class AddTaskWizardComponent implements OnInit, OnDestroy {
...
+  public addService() {
+    const payload = {
+      name: this.taskForm.value['name'],
+      type: this.taskType.toLowerCase(),
+      plugin: this.taskForm.value['plugin'][0],
+      ...this.taskForm.value['config'] && { config: this.taskForm.value['config'] },
+      enabled: this.isScheduleEnabled
+    };
+
+    console.log('service ', payload);
```
Comment (praveen-garg) [inline src/app/components/core/north/add-task-wizard/add-task-wizard.component.ts:281]: remove?

File: src/app/components/core/north/add-task-wizard/add-task-wizard.component.html (hunk 4 above)
Comment (mshariq-nerd) [reply src/app/components/core/north/add-task-wizard/add-task-wizard.component.html:123]: yes

File: src/app/components/core/north/add-task-wizard/add-task-wizard.component.ts [hunk 8]
```diff
@@ -368,46 +323,35 @@ export class AddTaskWizardComponent implements OnInit, OnDestroy {
...
+      this.taskForm.controls['config'].patchValue(pluginInfo?.config);
+      this.taskForm.controls['config'].updateValueAndValidity({ onlySelf: true });
+      this.isScheduleEnabled = true; // set to default
+      this.cdRef.detectChanges();
     }
   }
 
   onCheckboxClicked(event) {
     this.isScheduleEnabled = event.target.checked ? true : false;
-    this.payload.schedule_enabled = this.isScheduleEnabled;
```
[the remark this answers is not in the document]
Comment (mshariq-nerd) [reply src/app/components/core/north/add-task-wizard/add-task-wizard.component.ts:410]: yes

File: src/app/components/core/north/add-task-wizard/add-task-wizard.component.html (hunk 5 above)
Comment (mshariq-nerd) [reply src/app/components/core/north/add-task-wizard/add-task-wizard.component.html:133]: yes

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**